File: 489C/main.cpp
#include <iostream>
#include <string>
using namespace std;

string find_min(int m, int s) {
    if (s == 0) {
        return (m == 1) ? "0" : "";
    }
    string res;
    int remaining = s;
    for (int i = 0; i < m; ++i) {
        int lower = (i == 0) ? 1 : 0;
        int upper = 9;
        bool found = false;
        for (int d = lower; d <= upper; ++d) {
            int max_remaining = (m - 1 - i) * 9;
            if (remaining - d >= 0 && remaining - d <= max_remaining) {
                res += to_string(d);
                remaining -= d;
                found = true;
                break;
            }
        }
        if (!found) {
            return "";
        }
    }
    return res;
}

string find_max(int m, int s) {
    if (s == 0) {
        return (m == 1) ? "0" : "";
    }
    string res;
    int remaining = s;
    for (int i = 0; i < m; ++i) {
        int upper = 9;
        int lower = (i == 0) ? 1 : 0;
        bool found = false;
        for (int d = upper; d >= lower; --d) {
            int max_remaining = (m - 1 - i) * 9;
            if (remaining - d >= 0 && remaining - d <= max_remaining) {
                res += to_string(d);
                remaining -= d;
                found = true;
                break;
            }
        }
        if (!found) {
            return "";
        }
    }
    return res;
}

int main() {
    int m, s;
    cin >> m >> s;

    if (s == 0) {
        if (m == 1) {
            cout << "0 0" << endl;
        } else {
            cout << "-1 -1" << endl;
        }
    } else {
        if (s > 9 * m) {
            cout << "-1 -1" << endl;
        } else {
            string min_num = find_min(m, s);
            string max_num = find_max(m, s);
            if (min_num.empty() || max_num.empty()) {
                cout << "-1 -1" << endl;
            } else {
                cout << min_num << " " << max_num << endl;
            }
        }
    }
    return 0;
}

File: 489C/main.py
m, s = map(int, input().split())

def find_min(m, s):
    if s == 0:
        return "0" if m == 1 else None
    res = []
    remaining = s
    for i in range(m):
        if i == 0:
            lower = 1
        else:
            lower = 0
        upper = 9
        for d in range(lower, upper + 1):
            max_remaining = (m - 1 - i) * 9
            if remaining - d >= 0 and remaining - d <= max_remaining:
                res.append(str(d))
                remaining -= d
                break
        else:
            return None
    return ''.join(res)

def find_max(m, s):
    if s == 0:
        return "0" if m == 1 else None
    res = []
    remaining = s
    for i in range(m):
        upper = 9
        if i == 0:
            lower = 1
        else:
            lower = 0
        for d in range(upper, lower - 1, -1):
            if remaining - d >= 0 and remaining - d <= (m - 1 - i) * 9:
                res.append(str(d))
                remaining -= d
                break
        else:
            return None
    return ''.join(res)

if s == 0:
    if m == 1:
        print("0 0")
    else:
        print("-1 -1")
else:
    if s > 9 * m:
        print("-1 -1")
    else:
        min_num = find_min(m, s)
        max_num = find_max(m, s)
        if min_num is None or max_num is None:
            print("-1 -1")
        else:
            print(min_num, max_num)